contributors/design-proposals/pod-priority-resourcequota.md

```diff
@@ -0,0 +1,77 @@
+
+## Overview 
+
+This design doc introduces the motivation of adding priority filed in ResourceQuota and
+how this priority impacts ResourceQuta applying to Pod. 
+
+## Detailed Design 
+
+### Expected behavior of ResourceQuota apply to pod 
+
+The design will be like:
+
+```go
+// ResourceQuotaSpec defines the desired hard limits to enforce for Quota
+type ResourceQuotaSpec struct {
+  // Hard is the set of desired hard limits for each named resource
+  // +optional
+  Hard ResourceList
+  // A collection of filters that must match each object tracked by a quota.
+  // If not specified, the quota matches all objects.
+  // +optional
+  Scopes []ResourceQuotaScope
+  // If specified, indicates this ResourceQuota applies to that priority class
+  // If not specified, this ResourceQuota applies to all priority classes.
+  // +optional
+  PriorityClassName string
+}
+```
+PriorityClassName in ResourceQuota is matched against PriorityClassName in the Pod, including the case where PriorityClassName is not set. In detail:
+
+* One could expect when `PriorityClassName` field is set, it indicates that this quota applies to to pods with the same priority class.
+* If this field is not present, it indicates this quota matches to pods that do not have a PriorityClassName (Pods' integer priorities would be ignored).
+
+It is worth noting that quota is assigned per namespace, not per user, in Kubernetes. Adding priority class name to ResourceQuota will not change anything with respect to the scope of ResourceQuota. In other words, when priority class name is defined for a ResourceQuota object, the quota is applied to all pods at that priority class in the specified namespace.
+
+Unlike pod spec, ResourceQuota does not have Priority *int32 in its spec.
+As explained above, ResourceQuota with a PriorityClassName matches against pods with the same PriorityClassName, regarless of the integer value of priority.
+
+### Expected behavior of ResourceQuota admission controller
+
+This part is about how to make admission controller work with priority aware `ResourceQuota`.
+
+* ResourceQuota admission controller should reject creation or update of ResourceQuota object if new ResourceQuota names a PriorityClass that does not exist, i.e. is not a pre-defined or user-specified PriorityClass (but empty is allowed).
+* ResourceQuota admission controller should reject creation of ResourceQuota object if another ResourceQuota object with same PriorityClass already exists.
+* ResourceQuota admission controller should reject update of ResourceQuota object if the new (updated) PriorityClass matches that of another ResourceQuota object.
+* priority admission controller should reject deletion of a PriorityClass object if it is named in any ResourceQuota object in any namespace.
```

Comment:
I am not so sure if we actually need this rule. We decided to not perform such a check when deleting a PriorityClass that is specified in existing pods. When discussing whether we needed the check, I was told that Kubernetes does not have referential integrity. For the same reason, and the fact that in the current design of ResourceQuota, we only match the given PriorityClassName against pods' PriorityClassNames and do not need the corresponding PriorityClass object to be present.